src/main/java/gov/nasa/pds/registry/common/es/dao/DataLoader.java

```diff
@@ -302,15 +316,20 @@ public int loadBatch(List<String> data, Set<String> errorLidvids) throws Excepti
             // Get HTTP response code
             int respCode = getResponseCode(con);
             if(respCode <= 0) throw ex;
-            
-            // Try extracting JSON from multi-line error response (last line) 
+
+            // Try extracting JSON from multi-line error response (last line)
             String json = DaoUtils.getLastLine(con.getErrorStream());
             if(json == null) throw ex;
-            
+
             // Parse error JSON to extract reason.
             String msg = EsUtils.extractReasonFromJson(json);
             if(msg == null) msg = json;
-            
+
+            if (retries > 0) {
```

Comment:
I am missing a `retries -= 1;` somewhere before this line ?

It does not sound too complicated also to add a sleep(2**(1+num_retry)), so that max wait 2**6 = 64 in seconds, and that might be worth it since otherwise we will have all the requests launched back and forth without giving much chances for the system to recover from any short outage.

You would need to have an additional argument in your method to manage the max_num of retries.

Reply:
@alexdunnjpl , I did not read that https://harish-bhattbhatt.medium.com/best-practices-for-retry-pattern-f29d47cd5117 , could you go through that ?

Reply:
Ugh, good catch, thank-you - I performed a "minor" refactor right after testing that lost the decrement in the recursive call.  Will fix.  

Regarding retries, the logs show single requests timing out (with immediately-subsequent requests succeeding), hence the decision not to incorporate backoff (yet).

On that basis, I'm inclined to save expo backoff until we implement a proper solution, but I can add it if you think that's undesirable.

Reply:
@tloubrieu-jpl fixed here

https://github.com/NASA-PDS/registry-common/pull/42/files#diff-44a3fe7c92e6be4e43f1bd99836e8af4331ce231e3d2bfda2a9f7f26a63a1e29R330

Reply:
Hi @alexdunnjpl , that sounds good to me